src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -53,9 +51,59 @@ class HttpGoogleCloudStorageDAO(
 
   val system = akka.actor.ActorSystem("system")
 
-  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String): Unit = {
-    val inserter = getStorage(getBucketCredential(userInfo)).buckets.insert(projectId, new Bucket().setName(getBucketName(workspaceId)))
-    retry(inserter.execute,when500)
+  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Unit = {
+    val bucketName = getBucketName(workspaceId)
+    val groups = getGroupDirectory.groups
+    insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Read)) {
```

Comment:
Would you feel better about it if I called the method "withGroup" rather than "insertGroup" so that it suggests the pattern that's being used?